test/test_ur_urdf_xacro.py

```diff
@@ -0,0 +1,111 @@
+# Copyright (c) 2022 FZI Forschungszentrum Informatik
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+#    * Redistributions of source code must retain the above copyright
+#      notice, this list of conditions and the following disclaimer.
+#
+#    * Redistributions in binary form must reproduce the above copyright
+#      notice, this list of conditions and the following disclaimer in the
+#      documentation and/or other materials provided with the distribution.
+#
+#    * Neither the name of the {copyright_holder} nor the names of its
+#      contributors may be used to endorse or promote products derived from
+#      this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+#
+# Author: Lukas Sackewitz
+
+import os
+import shutil
+
+from ament_index_python.packages import get_package_share_directory
+
+
+def test_ur_urdf_xacro():
+
+    # Initialize Arguments
+    ur_type = "ur3"
+    safety_limits = "true"
+    safety_pos_margin = "0.15"
+    safety_k_position = "20"
+    # General Arguments
+    description_package = "ur_description"
+    description_file = "ur.urdf.xacro"
+    prefix = '""'
+
+    joint_limit_params = os.path.join(
+        get_package_share_directory(description_package), "config", ur_type, "joint_limits.yaml"
+    )
+    kinematics_params = os.path.join(
+        get_package_share_directory(description_package),
+        "config",
+        ur_type,
+        "default_kinematics.yaml",
+    )
+    physical_params = os.path.join(
+        get_package_share_directory(description_package),
+        "config",
+        ur_type,
+        "physical_parameters.yaml",
+    )
+    visual_params = os.path.join(
+        get_package_share_directory(description_package),
+        "config",
+        ur_type,
+        "visual_parameters.yaml",
+    )
+
+    description_file_path = os.path.join(
+        get_package_share_directory(description_package), "urdf", description_file
+    )
+
+    urdf_tmp_output = os.path.join(
+        get_package_share_directory(description_package), "test", "tmp_ur.urdf"
```

Comment:
It might be nice to use [`mkstemp`](https://docs.python.org/3/library/tempfile.html#tempfile.mkstemp) here instead of creating files in the share directory.